torch/_dynamo/variables/tensor.py

```diff
@@ -744,74 +764,63 @@ def call_method(
             kwargs,
         )
 
-        # TODO(future PR): implement rewrapping conditional on method presence
-        # in `torch.overrides.get_default_nowrap_function()`. It's unclear how
-        # to do this easily in the current codebase since the resolution of
-        # `GetAttrVariable` depends on the type of the underlying object.
-
-        return TensorWithTFOverrideVariable(
-            unwrapped,
-            self.orig_tensor_variable_source,
-            self.subclass_torch_function__func,
-            self.subclass_type,
-        )
-
     def global_class_name(self):
         return f"__subclass_{self.subclass_type.__name__}"
 
     @staticmethod
-    def inline_torch_function_unwrapped(
-        tx,
-        original_func_var,
-        tensor_with_tf_override_source,
-        tf_func,
-        subclass_type,
-        options,
-        args,
-        kwargs,
-    ):
-        """
-        This function inlines the `__torch_function__` override for `original_func_var`.
-        For example, if the user code is
+    def can_dispatch_torch_function(tx, args, kwargs):
+        if tx.output.torch_function_enabled:
+            all_args = args + tree_flatten(kwargs)[0]
+            return any(
+                isinstance(arg, TensorWithTFOverrideVariable) for arg in all_args
```

Comment:
I want to point out that this isn't entirely accurate. For C API bound functions, we typically only test for `__torch_function__` override on Tensor arguments. See also https://github.com/pytorch/pytorch/issues/53937

Reply:
You asked me if we could land this, even if it was somewhat unsound. I'm amenable to this. But we must make our best effort to thoroughly document all the places where we are unsound, so that it is easy for people to tell what needs to be beefed up later.

Reply:
Agreed, I didn't know this, I can update this. How does that handle objects with tf overrides? are they ignored?

Reply:
If you pass a TF override object to a non-Tensor field it is silently ignored, likely because we raised an exception. Unfortunately, I don't know a low effort way to get full fidelity here. In principle we can reimplement the python arg parser logic from Dynamo to do this matching but you still need a way of understanding what any given torch function/method is actually going to end up argument parsing with. To make matters worse, we don't consistently inline so you might hit some functions that have to go into Python first, and for those the only proper way to handle things is to actually inline.

Reply:
Ok, I'll make a note of this and have dynamo do the "correct" thing - having it check all arguments. This will be inconsistent with eager, but better imo.